platform/linux-generic/odp_queue.c

```diff
@@ -192,6 +201,9 @@ static odp_queue_t queue_create(const char *name,
 		param = &default_param;
 	}
 
+	if (param->size > CONFIG_QUEUE_SIZE)
```

Comment:
Change to `if (param->size >= CONFIG_QUEUE_SIZE)` to handle the effective queue capacity. The user-supplied `size` should then be set to `ROUNDUP_POWER2_U32(size) - 1` for the masking to work properly.

Reply:
One entry is not lost. User provided size if not (currently) used. Queue size is always 4k.